specification/sbi_cove.adoc

```diff
@@ -393,25 +393,31 @@ This requires the host to make three separate ECALLs to the TSM:
 . `tee_host_tvm_initiate_fence()`
 . `sbi_covh_tvm_remove_pages()`
 
-Upon completion, the host may reclaim the confidential pages that were
-previously mapped in the region using
-`tee_host_tsm_reclaim_pages()`. The host must then continue the TVM execution
-and insert shared pages into
-the region using `tee_host_tvm_add_shared_pages()` on the page-fault when TVM
-tries to access the region. If
-the region of address space is unpopulated, the page removal ECALLs are not
-needed and the host can insert
-shared pages into the region on the next page-fault.
+Upon completion of the invalidation of references to confidential memory, the
+host may reclaim the confidential pages that were previously mapped in the
+region using `tee_host_tsm_reclaim_pages()`. The host must then continue the
+TVM execution and insert shared pages into the region using
+`tee_host_tvm_add_shared_pages()` on the page-fault when TVM tries to access
+the region. If the region of address space is unpopulated, the page removal
+ECALLs are not needed and the host can insert shared pages into the region on
+the next page-fault.
 
 The calling TVM vCPU is considered blocked until the assignment-change is
-completed. Attempts to run it
-with `sbi_covh_run_tvm_vcpu()` will fail. Any guest page faults taken by other
-TVM vCPUs in the
-invalidated pages continue to be reported to the host.
+completed. Attempts to run it with `sbi_covh_run_tvm_vcpu()` will fail.
+Any guest page faults taken by other TVM vCPUs in the invalidated pages
+continue to be reported to the host.
+
+Note that the TVM vCPU is blocked until the host completes the conversion to
+shared memory - this sequence may happen in two parts - invalidation of
+references to confidential memory (and address translation cache flushes if any)
+and, the addition of the mapping to shared memory - the host may run the TVM
+vCPU after the first part, and lazily handle the fault for the second part.
+Also the reclamation is of the confidential pages, and the shared memory pages
+provided by the host may be unique from those pages so that host has the option
+to service the request on the TVM synchronously or asynchronously.
```

Comment:
@ozkoyuncu this section has the main text added.

Reply:
ack. thx